hil/ext/switches/dellnos9.py

```diff
@@ -0,0 +1,377 @@
+        response = self._make_request('GET', url)
+        root = etree.fromstring(response.text)
+        shutdown = root.find(self._construct_tag('shutdown')).text
+        # is error handling needed here?
+        return (shutdown == 'false')
+
+    # HELPER METHODS *********************************************
+
+    def _execute(self, interface, command_type, command):
+        """This method gets the url & the payload and executes <command>"""
+        url = self._construct_url()
+        payload = self._make_payload(command_type, command)
+        return self._make_request('POST', url, data=payload)
+
+    def _construct_url(self, interface=None):
+        """ Construct the API url for a specific interface.
+
+        Args:
+            interface: interface to construct the url for
+
+        Returns: string with the url for a specific interface and operation
+
+        If interface is None, then it returns the URL for REST API CLI.
+        """
+
+        if interface is None:
+            return '%s/api/running/dell/_operations/cli' % self.hostname
+
+        val = re.compile(r'^\d+/\d+(/\d+)?$')
+        if val.match(interface):
+            # if `interface` refers to port name
+            # the urls have dashes instead of slashes in interface names
+            interface = interface.replace('/', '-')
+            interface_type = self._convert_interface(self.interface_type)
+        else:
+            # interface refers to `vlan`
+            interface_type = 'vlan-'
+
+        return ''.join([self.hostname, '/api/running/dell/interfaces/'
+                       'interface/', interface_type, interface])
+
+    @staticmethod
+    def _convert_interface(interface_type):
+        """ Convert the interface name from switch CLI form to what the API
+        server understands.
+
+        Args:
+            interface: the interface in the CLI-form
+
+        Returns: string interface
+        """
+        iftypes = {'GigabitEthernet': 'gige-',
+                   'TenGigabitEthernet': 'tengig-',
+                   'TwentyfiveGigabitEthernet': 'twentyfivegig-',
+                   'fortyGigE': 'fortygig-',
+                   'peGigabitEthernet': 'pegig-',
+                   'FiftyGigabitEthernet': 'fiftygig-',
+                   'HundredGigabitEthernet': 'hundredgig-'}
+
+        return iftypes[interface_type]
+
+    @property
+    def _auth(self):
+        return self.username, self.password
+
+    @staticmethod
+    def _make_payload(command, command_type):
+        """Makes payload for passing CLI commands using the REST API"""
+
+        return '<input><%s>%s</%s></input>' % (command, command_type, command)
+
+    @staticmethod
+    def _construct_tag(name):
+        """ Construct the xml tag by prepending the dell tag prefix. """
+        return '{http://www.dell.com/ns/dell:0.1/root}%s' % name
+
+    def _make_request(self, method, url, data=None,
+                      acceptable_error_codes=()):
```

Comment:
afaik, you don't use this parameter anywhere. More copypasta from the brocade driver.

In the future, please don't copy-and-paste; if there's enough in common to warrant it, there's enough in common to factor it out into its own code.